### PR TITLE
Update README for type embeds

### PR DESCRIPTION
**Pull request by surma**
I took a stab at updating the README for the new as-bind, in an effort to get a beta out for Wasm Summit :D

@torch2424 it does involve quite a bit of deletion as a lot of the properties are gone, but I think that’s a good thing! PTAL and let me know if you have any remarks or concerns!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,14 +33,13 @@ Isomorphic library to handle passing high-level data structures between Assembly
...
+- The library is isomorphic. Meaning it supports both the Browser, and Node! And has ESM, AMD, CommonJS, and IIFE bundles! 🌐
 - Wraps around the [AssemblyScript Loader](https://github.com/AssemblyScript/assemblyscript/tree/master/lib/loader). The loader handles all the heavy-lifting of passing data into WebAssembly linear memory. 💪
 - Wraps around imported JavaScript functions, and exported AssemblyScript functions of the AssemblyScript Wasm Module. This allows high-level data types to be passed directly to exported AssemblyScript functions! 🤯
-- The library works at runtime, so no generated code that you have to maintain and make it play nicely in your environment. 🏃
-- Maintains great performance (relative to generating the corresponding JavaScript code), by using [Speculative Execution](https://en.wikipedia.org/wiki/Speculative_execution), and caching types passed between functions. 🤔
+- Moves a lot of work to compile-time using [AssemblyScript Transforms](https://www.assemblyscript.org/transforms.html#transforms) and completely avoids module-specific “glue code”. 🏃
```
Comment (torch2424) [inline README.md:39]: :smile: :tada:

Comment (torch2424) [inline README.md:39]: :smile: :tada:

File: README.md [hunk 2]
```diff
@@ -33,14 +33,13 @@ Isomorphic library to handle passing high-level data structures between Assembly
...
 - Installable from package managers (npm), with a modern JavaScript API syntax. 📦
-- The library is [< 5KB (minified and gzip'd)](https://bundlephobia.com/result?p=as-bind) and tree-shakeable! 🌲
-- This library is currently (as of January, 2020) the [wasm-bindgen](https://github.com/rustwasm/wasm-bindgen) in the Rust/Wasm ecosystem, for AssemblyScript. 😀
+- The library is [< 4KB (minified and gzip'd)](https://bundlephobia.com/result?p=as-bind), _including_ the AssemblyScript Loader ! 🌲
```
Comment (torch2424) [inline README.md:41]: Oooo! Thanks for shaving off a KB :smile:

File: README.md [hunk 3]
```diff
@@ -33,14 +33,13 @@ Isomorphic library to handle passing high-level data structures between Assembly
...
+- This library is currently (as of January, 2020) the [wasm-bindgen](https://github.com/rustwasm/wasm-bindgen) of AssemblyScript. 😀
```
Comment (torch2424) [inline README.md:42]: :+1:

File: README.md [hunk 4]
```diff
@@ -52,15 +51,14 @@ You can install as-bind in your project by running the following:
...
 
-`asc ./node_modules/as-bind/lib/assembly/as-bind.ts your-entryfile.ts --runtime incremental --exportRuntime [...other cli options...]`
+`asc your-entryfile.ts --exportRuntime --transform as-bind [...other cli options...]`
 
 The things to notice are:
 
-- `./node_modules/as-bind/lib/assembly/as-bind.ts` - This is the as-bind entryfile, used for exporting IDs of AssemblyScript classes so we can use them for instantiating new classes
-- `--runtime incremental` - This specifies that we are using the incremental runtime / garbage collection option (The AssemblyScript default). However, [all the runtime options](https://www.assemblyscript.org/garbage-collection.html) are supported (incremental, minimal, and stub).
-- `--exportRuntime` - This allows us to use the [AssemblyScript Garbage Collection functions added in 0.18.x](https://www.assemblyscript.org/garbage-collection.html)
+- `--transform as-bind` - This is the as-bind transform that runs at compile time. It embeds all the required type information into the WebAssembly Module.
```
Comment (torch2424) [inline README.md:60]: Thanks for updating these! :smile:

File: README.md [hunk 5]
```diff
@@ -167,22 +165,9 @@ asyncTask();
 
 ## Supported Data Types
 
-**TL;DR:** Currently Numbers, Strings, Typed Arrays, and common Array<T> types are supported. Returning a high-level data type from an imported JavaScript function, and better AssemblyScript Class support will be coming later. For more detailed type support information, please see the [`lib/asbind-instance/supported-ref-types.js`](./lib/asbind-instance/supported-ref-types.js) and [`test/assembly/test.ts`](./test/assembly/test.ts).
+All primitive types, ie. Numbers (`u8`, `f32`, ...) , Strings, Typed Arrays (`Uint8Array`, `Float32Array`, ...) are supported. All of those types can also be used with `Array<T>`.
```
Comment (torch2424) [inline README.md:168]: Awesome! I think we may want to be more clear on what is not supported (That way we don't get issues opened if we do support it).

But I don't think it should be blocking :smile:

File: README.md [hunk 6]
```diff
@@ -314,27 +243,15 @@ This library was inspired by several chats I had with some awesome buddies of mi
...
 
 ## Performance
 
-**TL;DR** This library should be fast, but depending on your project you may want some more careful consideration. 🤔
+**TL;DR** This library should be pretty darn fast. 🤔
 
-as-bind does all of its data passing at runtime. Meaning this will be slower than a code generated bindings generator, such as something like [wasm-bindgen](https://github.com/rustwasm/wasm-bindgen). This is because, as-bind needs to cycle through every supported type on every paremeter or return value for each function, whenever the function is called. However, this is mitigated due to the Speculative execution that the library implements.
+The transform embeds all the required type information of imported and exported functions into a custom section of the WebAssembly module. All the runtime does is utilize the AssemblyScript Loader to convert these types from JS to ASC and vice-versa. Apart from `Array<T>`, which needs to be handled recursively, the overhead is fairly static and minimal.
```
Comment (torch2424) [inline README.md:252]: Perfect :sob: :smile: :tada:

**Review by torch2424 — APPROVED** [review]
LGTM! :smile: :ship:

**Comment by torch2424** [comment]
@surma I think you have the rights to merge? :smile: Feel free to do so!
